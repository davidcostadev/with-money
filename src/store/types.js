export const ADD_USER = 'ADD_USER';
export const TO_CURRENT_MONTH = 'TO_CURRENT_MONTH';
export const TO_PREVIUS_MONTH = 'TO_PREVIUS_MONTH';
export const TO_NEXT_MONTH = 'TO_NEXT_MONTH';
