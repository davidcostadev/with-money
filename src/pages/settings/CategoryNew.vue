<template>
  <div>
    <md-toolbar>
      <router-link to="/settings/categories">
        <md-button class="md-icon-button">
          <md-icon>keyboard_backspace</md-icon>
        </md-button>
      </router-link>
      <h3 class="md-title">Nova Categoria</h3>
    </md-toolbar>
    <category-form :onSave="onSave"/>
  </div>
</template>

<script>
import CategoryForm from '../../components/forms/Category';
import Categories from '../../services/categories';

export default {
  components: {
    CategoryForm,
  },
  methods: {
    async onSave(data) {
      const category = await Categories.create(data);

      if (category.id) {
        this.$store.dispatch('showFlashMessage', 'Categoria salva com sucesso!');
      }

      this.$router.push({
        path: '/settings/categories',
      });
    },
  },
};
</script>
