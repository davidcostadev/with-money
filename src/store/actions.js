import * as types from './types';

export default {
  addUser({ commit }, payload) {
    commit(types.ADD_USER, payload);
  },
  toNextMonth({ commit }) {
    commit(types.TO_NEXT_MONTH);
  },
  toCurrentMonth({ commit }) {
    commit(types.TO_CURRENT_MONTH);
  },
  toPreviousMonth({ commit }) {
    commit(types.TO_PREVIUS_MONTH);
  },
};
