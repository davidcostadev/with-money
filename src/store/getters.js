export default {
  user(state) {
    return state.user;
  },
  current_month(state) {
    return state.current_month;
  },
  state_month(state) {
    return state.state_month;
  },
  state_month_str(state) {
    return state.state_month_str;
  },
  next_month_str(state) {
    return state.next_month_str;
  },
  previous_month_str(state) {
    return state.previous_month_str;
  },
};
