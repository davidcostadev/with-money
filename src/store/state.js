import Moment from 'moment';

export default {
  user: null,
  current_month: Moment(),
  state_month: Moment(),
  state_month_str: Moment().format('MMM YY'),
  next_month_str: Moment().add(1, 'month').format('MMM YY'),
  previous_month_str: Moment().subtract(1, 'month').format('MMM YY'),
  flash_message: '',
};
