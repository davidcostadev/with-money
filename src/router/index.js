import Vue from 'vue';
import Router from 'vue-router';
import Login from '@/pages/Login';
import TransactionNew from '@/pages/TransactionNew';
import TransactionEdit from '@/pages/TransactionEdit';
import TransactionList from '@/pages/TransactionList';
import Reports from '@/pages/Reports';
import Settings from '@/pages/Settings';
import Accounts from '@/pages/settings/Accounts';
import AccountNew from '@/pages/settings/AccountNew';
import AccountEdit from '@/pages/settings/AccountEdit';
import Categories from '@/pages/settings/Categories';
import CategoryNew from '@/pages/settings/CategoryNew';
import CategoryEdit from '@/pages/settings/CategoryEdit';
import store from '../store';

Vue.use(Router);

const insecure = (to, from, next) => {
  const token = localStorage.getItem('token');

  if (!store.getters.user && !token) {
    next();
    return;
  }

  next('/');
};

const secure = (to, from, next) => {
  const token = localStorage.getItem('token');
  const user = localStorage.getItem('user');

  store.dispatch('addUser', JSON.parse(user));

  if (store.getters.user || token) {
    next();
    return;
  }

  next('/login');
};

export default new Router({
  routes: [
    {
      path: '/login',
      name: 'login',
      component: Login,
      beforeEnter: insecure,
    },
    {
      path: '/',
      name: 'TransactionList',
      component: TransactionList,
      beforeEnter: secure,
    },
    {
      path: '/transactions/:id',
      name: 'TransactionEdit',
      component: TransactionEdit,
      beforeEnter: secure,
    },
    {
      path: '/transaction-new',
      name: 'TransactionNew',
      component: TransactionNew,
      beforeEnter: secure,
    },
    {
      path: '/reports',
      name: 'Reports',
      component: Reports,
      beforeEnter: secure,
    },
    {
      path: '/settings',
      name: 'Settings',
      component: Settings,
      beforeEnter: secure,
    },
    {
      path: '/settings/accounts',
      name: 'Accounts',
      component: Accounts,
      beforeEnter: secure,
    },
    {
      path: '/settings/accounts/new',
      name: 'AccountNew',
      component: AccountNew,
      beforeEnter: secure,
    },
    {
      path: '/settings/accounts/edit/:id',
      name: 'AccountEdit',
      component: AccountEdit,
      beforeEnter: secure,
    },
    {
      path: '/settings/categories',
      name: 'Categories',
      component: Categories,
      beforeEnter: secure,
    },
    {
      path: '/settings/categories/new',
      name: 'CategoryNew',
      component: CategoryNew,
      beforeEnter: secure,
    },
    {
      path: '/settings/categories/edit/:id',
      name: 'CategoryEdit',
      component: CategoryEdit,
      beforeEnter: secure,
    },
  ],
});
