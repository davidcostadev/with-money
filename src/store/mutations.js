import Moment from 'moment';
import * as types from './types';
import { updateStatesMont } from '../utils/state';

export default {
  [types.ADD_USER](state, payload) {
    state.user = payload;
  },
  [types.TO_CURRENT_MONTH](state) {
    state.state_month = state.current_month;
    updateStatesMont(state);
  },
  [types.TO_PREVIUS_MONTH](state) {
    state.state_month = Moment(state.state_month).subtract(1, 'month');
    updateStatesMont(state);
  },
  [types.TO_NEXT_MONTH](state) {
    state.state_month = Moment(state.state_month).add(1, 'month');
    updateStatesMont(state);
  },
};
