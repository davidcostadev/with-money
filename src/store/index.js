import Vue from 'vue';
import Vuex from 'vuex';
import Moment from 'moment';
import mutations from './mutations';
import actions from './actions';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    user: null,
    current_month: Moment(),
    state_month: Moment(),
    state_month_str: Moment().format('MMMM YY'),
    next_month_str: Moment().add(1, 'month').format('MMMM YY'),
    previous_month_str: Moment().subtract(1, 'month').format('MMMM YY'),
  },
  getters: {
    user(state) {
      return state.user;
    },
    current_month(state) {
      return state.current_month;
    },
    state_month(state) {
      return state.state_month;
    },
    state_month_str(state) {
      return state.state_month_str;
    },
    next_month_str(state) {
      return state.next_month_str;
    },
    previous_month_str(state) {
      return state.previous_month_str;
    },
  },
  mutations,
  actions,
});
