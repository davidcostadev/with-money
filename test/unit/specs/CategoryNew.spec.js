import { createLocalVue, shallowMount } from '@vue/test-utils';
import VueMaterial from 'vue-material';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import CategoryNew from '../../../src/pages/settings/CategoryNew';
import store from '../../../src/store';
import router from '../../../src/router';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueMaterial);
localVue.use(VueRouter);

describe('CategoryNew Component', () => {
  beforeAll(() => {
    global.localStorage = {
      getItem: jest.fn().mockReturnValue(JSON.stringify({
        id: '1',
        name: 'name',
      })),
    };
  });

  describe('Render', () => {
    it('default', () => {
      const wrapper = shallowMount(CategoryNew, { store, localVue, router });

      expect(wrapper.isVueInstance).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
