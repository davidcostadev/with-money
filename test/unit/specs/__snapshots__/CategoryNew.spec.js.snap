// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryNew Component Render default 1`] = `
<div>
  <mdtoolbar-stub>
    <router-link-stub>
      <mdbutton-stub
        class="md-icon-button"
      >
        <mdicon-stub>
          keyboard_backspace
        </mdicon-stub>
      </mdbutton-stub>
    </router-link-stub>
     
    <h3
      class="md-title"
    >
      Nova Categoria
    </h3>
  </mdtoolbar-stub>
   
  <categoryform-stub />
</div>
`;
